Microsoft.Toolkit.Uwp/IncrementalLoadingCollection/IncrementalLoadingCollection.cs

```diff
@@ -226,8 +228,27 @@ public Task RefreshAsync()
         /// </returns>
         protected virtual async Task<IEnumerable<IType>> LoadDataAsync(CancellationToken cancellationToken)
         {
-            var result = await Source.GetPagedItemsAsync(CurrentPageIndex++, ItemsPerPage, cancellationToken);
-            return result;
+            await _mutex.WaitAsync();
+            try
```

Comment:
FYI @Sergio0694 added a helper [here](https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/main/Microsoft.Toolkit.Uwp.UI.Media/Extensions/System.Threading.Tasks/AsyncMutex.cs), but it's in the Media package. Just a note for the future, not sure if we want a TODO here for later. Not worth trying to centralize or expose now.